imap/jmap_mail.c

```diff
@@ -7236,6 +7236,26 @@ static int _email_get_bodies(jmap_req_t *req,
         json_object_set_new(email, "calendarEvents", calendar_events);
     }
 
+    /* previousCalendarEvent -- non-standard */
+    if (jmap_wantprop(props, "previousCalendarEvent")) {
+        json_t *event = NULL;
+        struct buf buf = BUF_INITIALIZER;
+        // cost is "load message body", so fetch with bodies
+        message_get_field(msg->_m, "X-ME-Cal-Previous", MESSAGE_RAW, &buf);
+        if (buf_len(&buf)) {
+            size_t datalen = ((4 * buf_len(&buf) / 3) + 3) & ~3;
```

Comment:
Can we plug that in a `base64len` inline function in charset.c or the like? We are doing these calculations at a couple of places and could later unify them all.

Reply:
It was all bogus so I removed it :)